script/commands/BridgeAsset.s.sol

```diff
@@ -0,0 +1,109 @@
+ *
+ * @dev How it works:
+ * 1. User provides destination chain ID via prompt
+ * 2. If on base chain (L1):
+ *    - Wraps ETH to WETH by sending ETH to WETH contract
+ * 3. Bridges tokens via OFT adapter's sendFrom()
+ *
+ * Usage:
+ * ```
+ * forge script script/commands/BridgeAsset.s.sol:BridgeAsset --rpc-url <RPC_URL> --broadcast
+ * ```
+ */
+contract BridgeAsset is BaseData {
+    using OptionsBuilder for bytes;
+
+    // Amount to bridge (0.1 ETH worth)
+    uint256 constant BRIDGE_AMOUNT = 0.12 ether;
+
+    function run() external {
+        uint256 destinationChainId =
+            vm.parseUint(vm.prompt("Enter destination chain ID (e.g. 2522 for Fraxtal testnet):"));
+        require(isSupportedChainId(destinationChainId), "Unsupported destination chain ID");
+        uint32 destinationEid = getEID(destinationChainId);
+
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+
+        address sender = vm.addr(deployerPrivateKey);
+        address refundAddress = sender;
+
+        console.log("Chain ID: %s", block.chainid);
+        console.log("Sender: %s", sender);
+        console.log("Destination Chain ID: %s", destinationChainId);
+        console.log("Destination EID: %s", destinationEid);
+
+        // Source chain ID
+        // If we're on Fraxtal testnet (2522), Morph testnet (2810), or Holesky (17000),
+        // use Holesky (17000) as the source chain since it's the L1 testnet.
+        // Otherwise default to Ethereum mainnet (1)
+        uint256 sourceChainId = block.chainid;
+        uint256 baseChainId = sourceChainId == 2522 || sourceChainId == 2810 || sourceChainId == 17000 ? 17000 : 1;
+
+        // Load deployment config
+        string memory json =
+            vm.readFile(string.concat("deployments/ynETHx-", vm.toString(baseChainId), "-v0.0.1.json"));
+
+        address oftAdapter = abi.decode(
+            vm.parseJson(json, string.concat(".chains.", vm.toString(sourceChainId), ".oftAdapter")), (address)
+        );
+
+        vm.startBroadcast(deployerPrivateKey);
+
+        // Get the ynETHx contract address
+        address ynETHx = abi.decode(
+            vm.parseJson(json, string.concat(".chains.", vm.toString(sourceChainId), ".erc20Address")), (address)
+        );
+
+        if (sourceChainId == baseChainId) {
+            // Get the WETH contract address from ynETHx
+            address weth = IERC4626(ynETHx).asset();
+
+            // Deposit ETH to get WETH
+            (bool success,) = weth.call{value: BRIDGE_AMOUNT}("");
+            require(success, "ETH to WETH deposit failed");
+
+            // Calculate amount of WETH needed to mint BRIDGE_AMOUNT of ynETHx
+            uint256 wethAmount = BRIDGE_AMOUNT;
+
+            // Approve WETH spending
+            IERC20(weth).approve(ynETHx, wethAmount);
+
+            // Deposit WETH to mint ynETHx using ERC4626 interface
+            IERC4626(ynETHx).deposit(wethAmount, sender);
```

Comment:
The shares returned from depositing wethAmount might not always be equal to BRIDGE_AMOUNT right ? @danoctavian